test/models/user_test.rb

```diff
@@ -0,0 +1,116 @@
+require "test_helper"
+
+describe User do
+  before do
+  end
+
+  describe 'relations' do
+    let(:user) { User.first }
+
+    it 'relates works and users' do
+      work = Work.first
+
+      user.works << work
+      user.work_ids.must_include work.id
+    end
+
+    it 'relates vote and vote_id' do
+      vote = Vote.create!(user_id: user.id, work_id: Work.first.id)
+
+      user.vote_ids.must_include vote.id
+    end
+  end
+
+  describe 'validations' do
+    before do
+      @user = User.new(username: 'Username')
+    end
+
+    it 'cannot be created without a username' do
+      @user.username = nil
+      result = @user.valid?
+
+      result.must_equal false
+      @user.errors.messages.must_include :username
+    end
+
+    it 'cannot be created if the username has been taken' do
+      # I know this situation will not happen on the website, but maybe can be useful if reading from a CSV or creating users through the console.
+      @user.save
+      username = @user.username
+      user2 = User.new(username: username)
+
+      result = user2.valid?
+
+      result.must_equal false
+    end
+
+    it 'can be created with all required fields' do
+      result = @user.valid?
+      result.must_equal true
+    end
+  end
+
+  describe 'recommender' do
+    before do
+      @tor = users(:tor)
+      @roo = users(:roo)
+      @hill = works(:hill)
+      @noscrubs = works(:noscrubs)
+
+      Vote.create!(user_id: @tor.id, work_id: @noscrubs.id)
+      Vote.create!(user_id: @roo.id, work_id: @noscrubs.id)
+      Vote.create!(user_id: @roo.id, work_id: @hill.id)
+
+    end
+
+    it 'must calculate a jaccard index between a two users' do
+      result = User.calculate_jaccard(@tor, @roo)
+
+      result.must_equal 0.5
```

Comment:
I'm going to start by appreciating that you have test coverage here at all - it's easy to let it slip when you're working on a science project like this.

That being said, the test coverage here could probably stand to be more broad. What if one of the users doesn't have any votes? What if there is no intersect, or if the intersect is the union? Does it work with many works?